core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,45 @@
+package io.udash.utils
+
+import java.io.IOException
+
+import org.scalajs.dom._
+import scalatags.JsDom.all._
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.typedarray._
+
+import scala.concurrent._
+import scala.util.Try
+
+object FileService {
+  def save(filename: String, bytes: Array[Byte], mimeType: String = "application/octet-stream"): Unit = {
+    val jsBytes = Array[js.Any](bytes.toTypedArray).toJSArray
+    val blob = new Blob(jsBytes, BlobPropertyBag(mimeType))
+    val url = URL.createObjectURL(blob)
+    val download = attr("download")
+
+    val anchor = a(href := url, download := filename, style := "display: none")(filename).render
+    document.body.appendChild(anchor)
+    anchor.click()
+    document.body.removeChild(anchor)
+    URL.revokeObjectURL(url)
```

Comment:
You may revoke here before the file is fully downloaded, which will truncate the download result for larger files / slower downloads.

Reply:
It is good question when to revoke it to be honest.

Reply:
I put an example with revoke to guide and added some comments about it. 

But I have no idea how to do it without leaking, and to reduce "leaking" I made a originally hack with `click` because it allows to obtain URL only when an application needs it.